### ISSUE TITLE
Trace shows negative number of inactive connections

### ISSUE THREAD

**Issue by jim2paterson** [reactor/reactor-netty#673]
### Expected behavior
Channel pool / connection management debug tracing shows only positive number of inactive connections. 

### Actual behavior
Early in the log, we see what looks like the 3 channels being created:

```
12:42:10.350 [nioEventLoopGroup-2-1] DEBUG r.n.r.PooledConnectionProvider - [id: 0x68a88c25] Created new pooled channel, now 0 active connections and 1 inactive connections
12:42:10.395 [nioEventLoopGroup-2-1] DEBUG r.netty.channel.BootstrapHandlers - [id: 0x68a88c25] Initialized pipeline DefaultChannelPipeline{(reactor.left.loggingHandler = io.netty.handler.logging.LoggingHandler), (BootstrapHandlers$BootstrapInitializerHandler#0 = reactor.netty.channel.BootstrapHandlers$BootstrapInitializerHandler), (SimpleChannelPool$1#0 = io.netty.channel.pool.SimpleChannelPool$1), (reactor.left.httpCodec = io.netty.handler.codec.http.HttpClientCodec), (reactor.left.decompressor = io.netty.handler.codec.http.HttpContentDecompressor), (reactor.right.reactiveBridge = reactor.netty.channel.ChannelOperationsHandler)}
12:42:10.396 [nioEventLoopGroup-2-1] DEBUG reactor.netty.tcp.TcpClient - [id: 0x68a88c25] REGISTERED
12:42:10.403 [nioEventLoopGroup-2-1] DEBUG r.n.r.PooledConnectionProvider - [id: 0x2ec5621d] Created new pooled channel, now 0 active connections and 2 inactive connections
12:42:10.404 [nioEventLoopGroup-2-1] DEBUG r.netty.channel.BootstrapHandlers - [id: 0x2ec5621d] Initialized pipeline DefaultChannelPipeline{(reactor.left.loggingHandler = io.netty.handler.logging.LoggingHandler), (BootstrapHandlers$BootstrapInitializerHandler#0 = reactor.netty.channel.BootstrapHandlers$BootstrapInitializerHandler), (SimpleChannelPool$1#0 = io.netty.channel.pool.SimpleChannelPool$1), (reactor.left.httpCodec = io.netty.handler.codec.http.HttpClientCodec), (reactor.left.decompressor = io.netty.handler.codec.http.HttpContentDecompressor), (reactor.right.reactiveBridge = reactor.netty.channel.ChannelOperationsHandler)}
12:42:10.404 [nioEventLoopGroup-2-1] DEBUG reactor.netty.tcp.TcpClient - [id: 0x2ec5621d] REGISTERED
12:42:10.405 [nioEventLoopGroup-2-3] DEBUG r.n.r.PooledConnectionProvider - [id: 0x64d8b385] Created new pooled channel, now 0 active connections and 3 inactive connections
```
Then a bit later we see the negative inactive connections:

```
12:42:10.721 [nioEventLoopGroup-2-1] DEBUG r.n.r.PooledConnectionProvider - [id: 0x68a88c25, L:/127.0.0.1:57407 - R:localhost/127.0.0.1:57405] Channel acquired, now 1 active connections and -44 inactive connections

12:42:10.726 [nioEventLoopGroup-2-1] DEBUG r.n.http.client.HttpClientOperations - [id: 0x68a88c25, L:/127.0.0.1:57407 - R:localhost/127.0.0.1:57405] Received last HTTP packet
12:42:10.726 [nioEventLoopGroup-2-1] DEBUG r.n.r.PooledConnectionProvider - [id: 0x68a88c25, L:/127.0.0.1:57407 - R:localhost/127.0.0.1:57405] onStateChange(POST{uri=/, connection=PooledConnection{channel=[id: 0x68a88c25, L:/127.0.0.1:57407 - R:localhost/127.0.0.1:57405]}}, [disconnecting])
12:42:10.726 [nioEventLoopGroup-2-1] DEBUG r.n.r.PooledConnectionProvider - [id: 0x68a88c25, L:/127.0.0.1:57407 - R:localhost/127.0.0.1:57405] Releasing channel
12:42:10.726 [nioEventLoopGroup-2-1] DEBUG r.n.r.PooledConnectionProvider - [id: 0x68a88c25, L:/127.0.0.1:57407 - R:localhost/127.0.0.1:57405] Channel cleaned, now 0 active connections and -43 inactive connections
```

### Steps to reproduce
Use the attached JUnit test and examine the DEBUG entries in the logs.  The test purposely exhausts the connection pool in order to exercise the error handling.

### Reactor Netty version
0.8.5.RELEASE

### JVM version (e.g. `java -version`)
openjdk version "11.0.2" 2019-01-15
but compiling for Java 9

### OS version (e.g. `uname -a`)
Darwin myhost 18.2.0 Darwin Kernel Version 18.2.0: Thu Dec 20 20:46:53 PST 2018; root:xnu-4903.241.1~1/RELEASE_X86_64 x86_64 i386 MacBookPro11,4 Darwin

This seems like a minor bug, but I discovered it trying to recreate a more problematic bug we see intermittently in production.  There we see a fixed connection pool that believes there are no items available even though all physical connections have been closed and there are no longer any open sockets. It seems to involve some combination of connection time-out errors and fixed connection pool exhausted (time-out on acquire operation for pool).  It may also involve time-outs on the servers HTTP response.  So far it has been difficult to come up with a simple recreation.  My hope is that maybe they are related.

**Comment by jim2paterson** [comment]
[test.zip](https://github.com/reactor/reactor-netty/files/3005774/test.zip)

**Comment by violetagg** [comment]
@jim2paterson The other issue may be related to #692

**Comment by violetagg** [comment]
Try the latest 0.8.7.BUILD-SNAPSHOT

**Comment by jim2paterson** [comment]
@violetagg It's definitely fixed in the latest 0.8.7.BUILD-SNAPSHOT.  Thanks!
